src/Modal.js

```diff
@@ -309,6 +315,8 @@ const Modal = React.createClass({
 
     this.setState(this._getStyles() //eslint-disable-line react/no-did-mount-set-state
       , () => this.focusModalContent());
+
+    this.needsFinalization = true;
```

Comment:
this is probably not needed, `onShow` is only ever called with props.show is set to true (either initially or updated) so checking if its true is enough to know if `onHide` needs to be called, which makes it consistent with `componentDidMount` as well